### PR TITLE
[WebNN EP] Remove workaround for CPU op supported list

### PR DESCRIPTION
**Pull request by Honry**
We assume all WebNN ops are supported across all backends.

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/helper.h [hunk 1]
```diff
@@ -249,14 +244,7 @@ inline bool CheckSingleOp(const std::string& op_type, const emscripten::val& wnn
...
   // Returns false if the WebNN op has not been implemented in MLGraphBuilder in current browser.
-  if (!wnn_builder[op_map.find(op_type)->second.opName].as<bool>()) {
-    return false;
-  }
-  // The current WebNN CPU (TFLite) backend supports a limited op list, and we'd rather
-  // fall back early to the ORT CPU EP rather than fail in the WebNN "cpu" deviceType.
-  // This is a workaround because the op may be included in MLGraphBuilder for DirectML
-  // backend but without TFLite implementation in Chromium.
-  if (!op_map.find(op_type)->second.isCpuSupported && device_type == WebnnDeviceType::CPU) {
+  if (!wnn_builder[op_map.find(op_type)->second].as<bool>()) {
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/helper.h:247]: Hmm, what does `InlinedHashMap` return if the key is missing? Is it like std::map, and if so, is `->second` valid if you pass say `op_type == "foo"`? Should it be this?

```
  auto op_map_entry = op_map.find(op_type);
  if (op_map_entry == op_map.end() || !wnn_builder[op_map_entry->second].as<bool>()) {
```

**Review by fdwr — APPROVED** [review]
Yay, glad to see this gone. Approved assuming the map lookup is correct.

File: onnxruntime/core/providers/webnn/builders/helper.h (hunk 1 above)
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/helper.h:247]:
```
  if (op_map.find(op_type) == op_map.end()) {
    return false;
  }
```
We've check if it contains the key at previous code.

But your code looks more concise, I will use it. :)

Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/helper.h]: Fixed, PTAL again.

Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/helper.h]: > We've check if it contains the key at previous code

The previous code that was deleted? 😉

Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/helper.h]: Ah, I missed it in the folded code.

**Review by fdwr — APPROVED** [review]
👍

File: onnxruntime/core/providers/webnn/builders/helper.h [hunk 2]
```diff
@@ -242,7 +242,7 @@ inline bool CheckSingleOp(const std::string& op_type, const emscripten::val& wnn
   auto op_map_entry = op_map.find(op_type);
   // Returns false if the op_type is not listed in the op_map or
   // if the WebNN op has not been implemented in MLGraphBuilder in current browser.
-  if (op_map_entry == op_map.end() || !wnn_builder[op_map_entry->second].as<bool()) {
+  if (op_map_entry == op_map.end() || !wnn_builder[op_map_entry->second].as<bool>()) {
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/builders/helper.h:245]: @fdwr, my bad, missing a `>`. :p

**Review by fdwr — APPROVED** [review]

**Pull request merged**